hi-ml-cpath/src/health_cpath/utils/montage.py

```diff
@@ -0,0 +1,517 @@
+        logging.info(f"Some image IDs contain '{escaped_amp}', replacing that with '&'.")
+        image_ids = [id.replace(escaped_amp, "&") for id in image_ids]
+    return pd.DataFrame({SlideKey.SLIDE_ID: image_ids, SlideKey.IMAGE: map(str, image_paths)})
+
+
+def restrict_dataset(dataset: pd.DataFrame,
+                     column: str,
+                     items: List[str],
+                     include: bool) -> pd.DataFrame:
+    """Exclude or include slides from a dataset, based on values in a column.
+    For example, to exclude slides with label `0` from the dataset, use:
+    restrict_dataset(dataset, column='label', items=['0'], include=False).
+
+    The items are matched with the column values using `isin` operator. The code also handles
+    the case where the column in question is the dataset index.
+    If the items in question are not present in the column, the result is an empty dataset (if include=True)
+    or the original dataset (if include=False).
+
+    :param dataset: Slides dataset.
+    :param column: The name of the column on which the inclusion/exclusion name.
+    :param items: The values that the column should match.
+    :param include: If True, modify the dataset to only include the rows where the column matches a value in the
+        item list in `items`. If False, modify the dataset to exclude those rows.
+    :return: Filtered dataset.
+    """
+    if column in dataset:
+        matching_rows = dataset[column].isin(items)
+        if not include:
+            matching_rows = ~matching_rows
+        return dataset[matching_rows]
+    elif dataset.index.name == column:
+        # Drop or loc operations on an index column when the values do not exist raise an error. Hence, restrict
+        # to existing values first.
+        items = list(set(items).intersection(set(dataset.index)))
+        if include:
+            return dataset.loc[items]
+        else:
+            return dataset.drop(items)
+    else:
+        raise ValueError(f"Column {column} not found in dataset.")
+
+
+def dataset_to_records(dataset: DatasetOrDataframe) -> List[DatasetRecord]:
+    """Converts a SlidesDataset or a plain dataframe to a list of dictionaries.
+
+    :param dataset: Slides dataset or a plain dataframe.
+    """
+    if isinstance(dataset, pd.DataFrame):
+        return dataset.to_dict(orient='records')
+    elif isinstance(dataset, SlidesDataset):
+        # SlidesData overrides __getitem__, use that to convert to records
+        return [dataset[i] for i in range(len(dataset))]
+    else:
+        raise ValueError(f"Can't convert {type(dataset)} to a list of records.")
+
+
+def make_montage(records: List[DatasetRecord],
+                 out_path: Path,
+                 width: int = 60_000,
+                 level: int = 2,
+                 image_suffix: str = '.png',
+                 masks: bool = True,
+                 temp_dir: Optional[Union[Path, str]] = None,
+                 cleanup: bool = False,
```

Comment:
are we able to set 1cleanup` from command line? If not, wondering if we want this as `False` by default?

Reply:
We are calling it with cleanup=True, so I think we are good.